internal/fdo_test.go

```diff
@@ -0,0 +1,204 @@
+package internal
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"fyne.io/desktop"
+	"fyne.io/fyne"
+	_ "fyne.io/fyne/test"
+	"github.com/magiconair/properties/assert"
+)
+
+var iconTheme = "default_theme"
+var iconSize = 32
+
+type dummyWindow struct {
+}
+
+type dummyWindow1 struct {
+	dummyWindow
+}
+
+type dummyWindow2 struct {
+	dummyWindow
+}
+
+type dummyWindow3 struct {
+	dummyWindow
+}
+
+type dummyWindow4 struct {
+	dummyWindow
+}
+
+func (*dummyWindow) Decorated() bool {
+	return true
+}
+
+func (*dummyWindow) Title() string {
+	return ""
+}
+
+func (*dummyWindow1) Title() string {
+	return "App1"
+}
+
+func (*dummyWindow) Class() []string {
+	return []string{"", ""}
+}
+
+func (*dummyWindow2) Class() []string {
+	return []string{"App2", "app2"}
+}
+
+func (*dummyWindow) Command() string {
+	return ""
+}
+
+func (*dummyWindow3) Command() string {
+	return "app3"
+}
+
+func (*dummyWindow) IconName() string {
+	return ""
+}
+
+func (*dummyWindow4) IconName() string {
+	return "app4"
+}
+
+func (*dummyWindow) Focus() {
+	// no-op
+}
+
+func (*dummyWindow) Close() {
+	// no-op
+}
+
+func (*dummyWindow) RaiseAbove(desktop.Window) {
+	// no-op (this is instructing the window after stack changes)
+}
+
+func exists(data desktop.IconData) bool {
+	var test bool
+	if data != nil && data.IconPath() != "" {
+		if _, err := os.Stat(data.IconPath()); err == nil {
+			test = true
+		}
+	}
+	return test
+}
+
+func setTestEnv() {
+	workingDir, err := os.Getwd()
+	if err != nil {
+		fyne.LogError("Could not get current working directory", err)
```

Comment:
Ignoring this error (or logging it) is bad - the test code around line 125 will then write files into the root of the filesystem